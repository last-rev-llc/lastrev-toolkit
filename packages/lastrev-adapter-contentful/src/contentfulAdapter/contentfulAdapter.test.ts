import Adapter from './contentfulAdapter';
import mockContent from './contentfulAdapter.mock';

const urlMap = {
  pageGeneral: {
    url: '/[key]',
    key: 'slug'
  }
};

const transform = Adapter({
  urlMap,
  linkContentType: 'uieCta',
  manualEntryTypeText: 'Manual URL',
  modalActionText: 'Open a modal',
  contentRefTypeText: 'ContentReference',
  skipContentTypes: ['skipThisType']
});

describe('Contentful Adapter', () => {
  it('converts the data correctly', () => {
    const out = transform(mockContent);
    expect(out).toMatchSnapshot();
  });
  it('parses plain json correctly', () => {
    const obj = {
      sys: {
        space: {
          sys: {
            type: 'Link',
            linkType: 'Space',
            id: 'hhv516v5f7sj'
          }
        },
        type: 'Entry',
        id: '1wmrGhKeCNbe3QssB6UNkN',
        contentType: {
          sys: {
            type: 'Link',
            linkType: 'ContentType',
            id: 'settingsGlobal'
          }
        },
        revision: 2,
        createdAt: '2020-07-01T20:03:32.191Z',
        updatedAt: '2020-07-06T23:50:58.713Z',
        environment: {
          sys: {
            id: 'master',
            type: 'Link',
            linkType: 'Environment'
          }
        },
        locale: 'en-US'
      },
      fields: {
        localizationLookup: {
          recipeMinutes: 'min',
          loginLinkLabel: 'LOG IN',
          loginDescription: 'Already have an Impossible Taste Placeâ„¢ account? Log in here.',
          logOutButtonLabel: 'LOG OUT',
          operatorLoginPreText: 'Restaurant operators',
          createAccountLinkLabel: 'Sign up for Taste Place here.',
          operatorLoginLinkLabel: 'log in here.',
          createAccountDescription: 'Earn rewards with each purchase. Save earth. Connect with other Impossible fans.'
        }
      }
    };
    const out = transform(obj);
    expect(out).toMatchSnapshot();
  });
});
