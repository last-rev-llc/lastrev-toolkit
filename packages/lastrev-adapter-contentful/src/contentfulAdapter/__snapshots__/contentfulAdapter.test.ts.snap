// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contentful Adapter converts the data correctly 1`] = `
Object {
  "_contentTypeId": "1234",
  "_id": "2345",
  "footerLogo": Object {
    "contentType": "image/png",
    "description": "nulla hic corporis",
    "filename": "image.png",
    "height": 70,
    "size": 231,
    "title": "Mock Image for footerLogo",
    "url": "//images.ctfassets.net/o8vbqs3spmqj/mock_settingsGlobal-footerLogo/460fedebd309366e91bbe10428654ab7/image.png",
    "width": 70,
  },
  "footerSocialLinks": Array [
    Object {
      "_contentTypeId": "uieCta",
      "_id": "37SQAoRR9kW7Qe9PVYPpXE",
      "as": null,
      "download": false,
      "href": "https://www.linkedin.com/company/uw-health",
      "icon": "Linkedin",
      "internalTitle": "LinkedIn social media CTA",
      "isModal": false,
      "linkText": "L",
      "target": "_blank",
    },
  ],
  "footerUtilityNavCta": Object {
    "_contentTypeId": "uieCta",
    "_id": "mock_UIE-CTA",
    "as": "/testslug",
    "download": false,
    "href": "/[slug]",
    "internalTitle": " Mock UIE-CTA",
    "isModal": true,
    "linkText": "quis explicabo ratione",
    "target": null,
  },
  "headerMainNavLogo": Object {
    "contentType": "image/png",
    "description": "natus voluptatem voluptatibus",
    "filename": "image.png",
    "height": 60,
    "size": 219,
    "title": "Mock Image for headerMainNavLogo",
    "url": "//images.ctfassets.net/o8vbqs3spmqj/mock_settingsGlobal-headerMainNavLogo/6597774d68edd42f663c5fb16ab0f43b/image.png",
    "width": 60,
  },
  "headerSearchErrorMessage": "There was an error",
  "headerSearchPlaceholderText": "Search",
  "headerUtilityNavLinks": Array [
    Object {
      "_contentTypeId": "elementNavLink",
      "_id": "mock_settingsGlobal-baselineCta",
      "_modifiedDate": "2020-06-17T19:39:13.522Z",
      "internalTitle": "Healthfacts - Nav Link",
      "mainLink": Object {
        "_contentTypeId": "uieCta",
        "_id": "mock_UIE-CTA",
        "as": null,
        "download": false,
        "href": "https://melody.biz",
        "internalTitle": " Mock UIE-CTA",
        "isModal": false,
        "linkText": "quis explicabo ratione",
        "target": null,
        "trackingId": "4321",
      },
    },
    Object {
      "_contentTypeId": "elementNavLink",
      "_id": "mock_settingsGlobal-baselineMainLinks",
      "_modifiedDate": "2020-06-17T19:39:26.301Z",
      "internalTitle": "Healthwise - Nav Link",
      "mainLink": Object {
        "_contentTypeId": "uieCta",
        "_id": "mock_UIE-CTA",
        "as": null,
        "download": true,
        "href": "https://melody.biz",
        "icon": "phone",
        "internalTitle": " Mock UIE-CTA",
        "isModal": false,
        "linkText": "quis explicabo ratione",
        "target": null,
        "trackingId": "9838",
      },
    },
    Object {
      "_contentTypeId": "elementNavLink",
      "_id": "mock_Element-NavLink",
      "_modifiedDate": "2020-06-17T19:42:15.908Z",
      "internalTitle": "Kids Health - Nav Link",
      "mainLink": Object {
        "_contentTypeId": "uieCta",
        "_id": "mock_UIE-CTA",
        "as": null,
        "download": false,
        "href": "//images.ctfassets.net/yadj1kx9rmg0/4gp6taAwW4CmSgumq2ekUm/9da0cd1936871b8d72343e895a00d611/Nyan_cat_250px_frame.png",
        "internalTitle": " Mock UIE-CTA",
        "isModal": false,
        "linkText": "quis explicabo ratione",
        "target": "_blank",
        "trackingId": "mollitia",
      },
    },
  ],
  "headerUtilityNavSiteNameLink": Object {
    "_contentTypeId": "elementNavLink",
    "_id": "mock_settingsGlobal-headerUtilityNavSiteNameLink",
    "_modifiedDate": "2020-06-17T20:00:03.043Z",
    "internalTitle": "Mock headerUtilityNavSiteNameLink for Settings - Global",
    "mainLink": Object {
      "_contentTypeId": "uieCta",
      "_id": "mock_UIE-CTA",
      "as": null,
      "download": false,
      "href": "#top",
      "icon": "",
      "internalTitle": " Mock UIE-CTA",
      "isModal": false,
      "linkText": "quis explicabo ratione",
      "style": "",
      "target": null,
      "trackingId": "mollitia",
    },
  },
  "internalTitle": "Patient Education Global Settings",
}
`;

exports[`Contentful Adapter parses plain json correctly 1`] = `
Object {
  "_contentTypeId": "settingsGlobal",
  "_id": "1wmrGhKeCNbe3QssB6UNkN",
  "_modifiedDate": "2020-07-06T23:50:58.713Z",
  "localizationLookup": Object {
    "createAccountDescription": "Earn rewards with each purchase. Save earth. Connect with other Impossible fans.",
    "createAccountLinkLabel": "Sign up for Taste Place here.",
    "logOutButtonLabel": "LOG OUT",
    "loginDescription": "Already have an Impossible Taste Placeâ„¢ account? Log in here.",
    "loginLinkLabel": "LOG IN",
    "operatorLoginLinkLabel": "log in here.",
    "operatorLoginPreText": "Restaurant operators",
    "recipeMinutes": "min",
  },
}
`;
